sunpy/instr/goes.py

```diff
@@ -583,16 +612,489 @@ def _goes_get_chianti_em(longflux, temp, satellite=8, abundances="coronal",
     modelflux = np.asarray(modelflux)
 
     # Ensure input values of flux ratio are within limits of model table
-    if np.min(np.log10(temp)) < np.min(modeltemp) or \
-      np.max(np.log10(temp)) > np.max(modeltemp) or \
-      np.isnan(np.min(np.log10(temp))):
+    if np.min(log10_temp) < np.min(modeltemp) or \
+      np.max(log10_temp) > np.max(modeltemp) or \
+      np.isnan(np.min(log10_temp)):
         raise ValueError("All values in temp must be within the range "
                          "{0} - {1} MK.".format(np.min(10**modeltemp),
                                                 np.max(10**modeltemp)))
 
     # Perform spline fit to model data
     spline = interpolate.splrep(modeltemp, modelflux, s=0)
-    denom = interpolate.splev(np.log10(temp), spline, der=0)
-    em = longflux/denom * 1e55
+    denom = interpolate.splev(np.log10(temp.value), spline, der=0)
+    em = longflux.value/denom * 1e55
+    em = Quantity(em, unit='cm**(-3)')
 
     return em
+
+def calculate_radiative_loss_rate(goeslc, force_download=False,
+                                  download_dir=DATA_PATH):
+    """
+    Calculates flare radiative loss rate and adds it to GOESLightCurve.
+
+    This function calculates the radiative loss rate as a function of
+    time of solar coronal soft X-ray-emitting plasma across all
+    wavelengths given a GOESLightCurve object.  The units of the
+    results are watts. This is done by calling _calc_rad_loss().
```

Comment:
Same thing here, about referencing a function that is hidden from the user. You should explain the functionality here.